compiler/rustc_codegen_llvm/src/back/target_machine_wrapper.rs

```diff
@@ -0,0 +1,98 @@
+use std::{
+    ffi::{c_char, CStr},
+    marker::PhantomData,
+    ops::Deref,
+    ptr::NonNull,
+};
+
+use rustc_data_structures::small_c_str::SmallCStr;
+
+use crate::{errors::LlvmError, llvm};
+
+#[repr(transparent)]
+pub struct TargetMachineWrapper {
```

Comment:
Not sure about the best name for this. I think `TargetMachineWrapper` isn't the best name, but it's also not bad. `TargetMachine` would be confusing. If you can think of something better, do rename it, but if you can't that's fine too and we can keep it.

Reply:
Some ideas might be
* TargetMachineBox (it kindo of behaves like a box, but its not copyable  )
* TargetMachineLLVM (it the the for the associated type TargetMachine for the LLVM backend.
* TargetMachineOwned / OwnedTargetMachine
* UniqueTargetMachine / TargetMachineUnique

Reply:
Renamed it to TargetMachineBox. I think that fits what I am going for and should good give a good intuition of what it does. Also clarified this in the comments.

Reply:
I like `OwnedTargetMachine` more than `TargetMachineBox`, but `TargetMachineBox` is fine too. If you want to, you can rename it again, or if you think it's fine and don't want to you can just use `@bors r=Nilstrieb` and approve the PR.